app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance_index = self._get_first_group_instance_index_in_block(group, blocks)
+            if first_group_instance_index is not None:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance_index, group['blocks'][0]['id'])
```

Comment:
I know it was already here but could you replace `group['blocks'][0]['id']` with `first_block_in_group`?

```
        for group in self.schema.groups:
            first_block_in_group = self.schema.get_first_block_id_for_group(group['id'])
            if not this_location:
                this_location = Location(group['id'], 0, first_block_in_group)

            ....

            first_group_instance_index = self._get_first_group_instance_index_in_block(group, blocks)
            if first_group_instance_index is not None:
                if group['id'] in first_groups:
                    this_location = Location(group['id'], first_group_instance_index, first_block_in_group)
```

Reply:
Done it , but needed an additional change , to make sure first_block_in_group set on each iteration of groups